### PR TITLE
feat: Add `ui.layout_columns()`

### LINKED ISSUE #855: Add the equivalent of `bslib::layout_columns()`
Fairly recently, `{bslib}` gained `layout_columns()`, which is now our recommended starting point for column-wise layout. 

https://rstudio.github.io/bslib/reference/index.html#multiple-columns

It also supersedes the the `row()`/`column()` interface (it supports filling layouts, wrapping onto new rows, etc), so when this gets added we should also drop `row()`/`column()` from express.

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #855
Pairs with rstudio/bslib#931

### CODE REVIEW

File: examples/layout_columns/app.py
Comment (cpsievert) [inline examples/layout_columns/app.py:34]: I think it's more important to show off the "intelligent default behavior" than it is to demonstrate what's possible (I can imagine a "drive by" reader could be off-put by the weird result without reading the source). 

This way, we can still give a hint as to an interesting customization, but have the result show-off the default behavior

```suggestion
            # col_widths=(-2, 4, 4, -2),
            # row_heights=["auto", 1]
```

File: shiny/express/layout.py
Comment (cpsievert) [inline shiny/express/layout.py:251]: I think we should be documenting `args` here?

File: shiny/express/layout.py
Comment (cpsievert) [inline shiny/express/layout.py:326]: I suppose this is consistent with `layout_column_wrap()`, but it's likely going to get removed for an alternative approach to detecting/setting intelligent page defaults, it'd be a heads-up move to remove this here and `layout_column_wrap()` 

```suggestion
```

File: shiny/ui/_layout_columns.py
Comment (cpsievert) [inline shiny/ui/_layout_columns.py:214]: Worth putting in a `bs_breakpoints()` (like we have in bslib?)

File: shiny/ui/_layout_columns.py
Comment (cpsievert) [inline shiny/ui/_layout_columns.py:166]: `layout_column_wrap()` should use this too?

Comment (cpsievert) [inline shiny/ui/_layout_columns.py:166]: `layout_column_wrap()` should use this too?

File: shiny/ui/_layout_columns.py
Comment (cpsievert) [inline shiny/ui/_layout_columns.py:203]: It'd be more readable for me if this func came immediately after `def layout_columns`

File: shiny/ui/_layout_columns.py
Comment (cpsievert) [inline shiny/ui/_layout_columns.py:256]: Is this getting used anywhere?

File: examples/layout_columns/app.py
Comment (gadenbuie) [reply examples/layout_columns/app.py:34]: Good point; the current state of this example is not intended as a user demonstration. Where do these examples end up and who is their intended audience? Currently it's aimed at testing and reviewing this PR, and we can certainly clean it up before merging.

File: shiny/express/layout.py
Comment (gadenbuie) [reply shiny/express/layout.py:251]: I'm not sure, it isn't documented in `shiny.express.layout_column_wrap()`, so I also didn't document it for `layout_columns()`.

https://github.com/posit-dev/py-shiny/blob/3bc470f9a3e4ca58749a5968cbbe3f5327d7825f/shiny/express/layout.py#L168-L170

File: shiny/ui/_layout_columns.py
Comment (gadenbuie) [reply shiny/ui/_layout_columns.py]: Autocomplete? I remove the useless alias

File: shiny/ui/_layout_columns.py
Comment (gadenbuie) [reply shiny/ui/_layout_columns.py]: Not any more, thanks!

File: shiny/ui/_layout_columns.py
Comment (gadenbuie) [reply shiny/ui/_layout_columns.py]: I rewrote `Breakpoints` to use an `Enum` instead so that the values are available for use.

File: shiny/express/layout.py
Comment (gadenbuie) [reply shiny/express/layout.py]: Sounds good, I removed this line! I'd like to keep the changes in this PR limited to the addition of `layout_columns()`. I'll be happy to follow up with a PR for `layout_column_wrap()`.

File: shiny/ui/_layout_columns.py
Comment (gadenbuie) [reply shiny/ui/_layout_columns.py:176]: Probably! This isn't a drop-in replacement for what happens to the children in  `layout_column_wrap()`, so I'd like to do this in a follow-up PR.

File: examples/layout_columns/app.py
Comment (cpsievert) [reply examples/layout_columns/app.py:34]: Ah, I was thinking that this would likely go under `shiny/api-examples/` which is where we put examples designed for the examples section of the API reference. I though that, in order to actually get the example in there, you need to decorate the function with `@add_example`, but it looks like that [isn't necessary for `layout_column_wrap`](https://shiny.posit.co/py/api/ui.layout_column_wrap.html) (btw, can you fix that `width="250%"` -> `width="250px"` in that example)? @schloerke probably knows more about how that works nowadays.

Also, the `./examples/` dir I think is more so intended for 'gallery'-like examples that combine many pieces at once (rather than one function in isolation).

Comment (gadenbuie) [reply examples/layout_columns/app.py:34]: Okay, in that case, to not hold up this PR any more than necessary, how about I take this example out before merging and add it in a follow up PR?

Comment (cpsievert) [reply examples/layout_columns/app.py:34]: 👍

Comment (gadenbuie) [reply examples/layout_columns/app.py]: 👉 https://github.com/posit-dev/py-shiny/issues/898

File: shiny/ui/_layout_columns.py
Comment (gadenbuie) [reply shiny/ui/_layout_columns.py:176]: 👉 https://github.com/posit-dev/py-shiny/issues/899

**Review by cpsievert — APPROVED** [review]

**Comment by gadenbuie** [comment]
Docs are going to be in a semi-broken state, but I have a fix planned and will follow up (#900)

**Pull request merged**
